genfstab: add page

peterbabic opened the pull request:
<!-- Thank you for sending a PR! -->
<!-- Please perform the following checks and mark all the boxes accordingly. -->
<!-- You can remove the checklist items that don't apply to your PR. -->

- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/tldr-pages/tldr?pullRequest=5215) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/tldr-pages/tldr?pullRequest=5215) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/tldr-pages/tldr?pullRequest=5215) it.</sub>

navarroaxel commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+# genfstab
+
+> Generate output suitable for addition to an fstab file.
+> More information: <https://man.archlinux.org/man/extra/arch-install-scripts/genfstab.8.en>.
```
```suggestion
> More information: <https://man.archlinux.org/man/extra/arch-install-scripts/genfstab.8>.
```

navarroaxel commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+
+`genfstab -L MOUNTPOINT`
+
+- Generate the output based on UUID:
```
Maybe this is better:
```suggestion
- Display `fstab` output using partitions' UUID:
```

navarroaxel commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+
+- Generate the output based on UUID:
+
+`genfstab -U MOUNTPOINT`
```
```suggestion
`genfstab -U {{path/to/mount_point}}`
```

navarroaxel commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+
+- Generate the output based on LABEL:
+
+`genfstab -L MOUNTPOINT`
```
```suggestion
`genfstab -L {{path/to/mount_point}}`
```

navarroaxel commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+
+- Generate a fstab file, requires being a superuser:
+
+`genfstab -U /mnt >> /mnt/etc/fstab`
```
```suggestion
`genfstab -U {{/mnt}} >> {{/mnt/etc/fstab}}`
```

navarroaxel commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+
+- Append a filesystem into fstab file to mount it automatically:
+
+`gengstab -U MOUNTPOINT | sudo tee -a /etc/fstab`
```
```suggestion
`genfstab -U {{path/to/mount_point}} | sudo tee -a /etc/fstab`
```

peterbabic reviewed (commented):
add maintainer suggestions

**EDIT:** do not know what to do, still reluctantly learning Github 
**EDIT2:** somehow aded maintainer suggestions into "batch" and commited them, hopefully this was expected

bl-ue commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+> Generate output suitable for addition to an fstab file.
+> More information: <https://man.archlinux.org/man/extra/arch-install-scripts/genfstab.8>.
+
+- Generate the output based on LABEL:
```
This is a little confusing&mdash;where does LABEL come from?

peterbabic replied:
The LABEL is a filesystem volume identifier, usually assigned (but not limited to) during the volume creation. More information: https://wiki.archlinux.org/index.php/Persistent_block_device_naming#by-label

bl-ue replied:
Then can you update either the description or the command, because it is not clear what "LABEL" has to do with the example command itself.

Maybe something like this?
```suggestion
- Generate the output based on a label:
```
and then change the example from `path/to/mount_point` to `path/to/label`?

I don't use Arch Linux, so feel free to correct me!

bl-ue commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+
+`genfstab -U {{path/to/mount_point}}`
+
+- Generate a fstab file, requires being a superuser:
```
Instead of giving superuser notice, you can just add `sudo` to the command (see additional suggestion below):
```suggestion
- Generate an fstab file:
```

peterbabic replied:
An fstab? I do not understand the edit.

bl-ue replied:
"an fstab file".  I assume "fstab" is generally pronounced "ef-es-tab"&mdash;@navarroaxel am I correct about this?

bl-ue replied:
It's better English to use "an" before a word like "fstab", the way it's generally pronounced.  Now of course, people have different pronunciations but I believe "an" would be a better fit here.
```suggestion
- Generate an fstab file, requires being a superuser:
```
Please also disregard the first suggestion, as I understand the superuser situation now :)

sbrl replied:
Yeah, that's correct grammar there @bl-ue.

bl-ue commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+
+- Generate a fstab file, requires being a superuser:
+
+`genfstab -U {{/mnt}} >> {{/mnt/etc/fstab}}`
```
See my suggestion above.
```suggestion
`sudo genfstab -U {{/mnt}} >> {{/mnt/etc/fstab}}`
```

peterbabic replied:
This suggestion will fail, for sudo there is the line just below.
Thiis line in specifically used when superuser and it is the 99% of it's use cases, I would definitely not edit it in this way.

bl-ue replied:
By "as superuser", do you mean after having run something like `su`?

bl-ue replied:
If it's required to run the command as superuser as you say, why not prefix it with `sudo`?  We do that in many other pages, particularly in low-level file-system utilities, such as [`partx.md`](https://github.com/tldr-pages/tldr/blob/master/pages/linux/partx.md).

peterbabic replied:
To your First question, yes, for instance `su` is necessary to elevate privileges.

To your second question, the command will fail because of the redirection operator `>>`. Prefixing the whole line with with `sudo` will not work. To use redirection operator with elevated privileges, a different approach is necessary.

https://linuxize.com/post/bash-append-to-file/#append-to-a-file-using-the-redirection-operator-
https://www.cyberciti.biz/faq/sudo-append-data-text-to-file-on-linux-unix-macos/

bl-ue replied:
Ah, I see!  Makes perfect sense now.  Please disregard this suggestion.

sbrl replied:
Hrm, I see the problem here. I would be inclined to go with @peterbabic's suggestion here, with something like this:

```suggestion
`genfstab -U {{path/to/mount_point}} | sudo tee -a /etc/fstab`
```

The reason for this is that by convention in other pages, we use `sudo` to indicate that a given command needs to be run as root - so I'd prefer to use that same convention here too if possible.

Also, welcome to GitHub and tldr-pages, @peterbabic!

bl-ue replied:
@sbrl this has already been resolved on another long thread on this same line :) - https://github.com/tldr-pages/tldr/pull/5215#discussion_r567993244

bl-ue reviewed (changes requested):
Hi @peterbabic, thank you for the new page! I've left some comments below for you to review.

peterbabic commented:
Unfortunately I do not agree wit any of the changes @bl-ue suggested, not meaning to be rude by any way. I have provided explanations to all of them.

bl-ue commented:
@peterbabic no problem! Sometimes we have difficulty communicating through a purely textual medium :)  I've responded to your comments.

bl-ue commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+
+- Generate an fstab file, requires being a superuser:
+
+`genfstab -U {{/mnt}} >> {{/mnt/etc/fstab}}`
```
How about this:
```suggestion
`genfstab -U {{path/to/mount_point}} >> {{/mnt/etc/fstab}}`
```

peterbabic replied:
I would rather change the description to: 

- Most usual way to generate an fstab file, requires being a superuser:

And left the code the way it is. What do you think?

bl-ue replied:
1. IMO, changing the description isn't necessary. I think it's good already 👍🏻 
2. `/mnt` is inconsistent it seems with the rest, that may be intentional (I'm no Arch user :smile:)

bl-ue replied:
Actually, I think it would be better to make this example more consistent with the last one, which would solve the superuser problem, by using `sudo tee`:
```suggestion
`genfstab -U {{path/to/mount_point}} | sudo tee -a {{/mnt/etc/fstab}}`
```

bl-ue replied:
If you're sure that `/mnt` is better, even though it's inconsistent, let it be that way. I trust you know better :)

navarroaxel replied:
@bl-ue in this example @peterbabic tried to explain the most common use case for this command as a step in the Arch Linux installation https://wiki.archlinux.org/index.php/installation_guide#Fstab. Nobody uses `sudo tee` 😄 because the installer is a live ISO and you run it with `root` user, command by command.

peterbabic replied:
Look, we can even discard the first and the second example. The third and the fourth are the only ones that I have personally ever needed and will serve for most people.

navarroaxel replied:
Yeah, but the first 2 examples are useful to explain how the command works and some arguments.

peterbabic replied:
Yes exactly, that is why I have added there. I was following the contribution guidelines.

bl-ue replied:
Okay guys, great!  @navarroaxel cool that you're an Arch Linux guy.  That's good to know.

> Nobody uses `sudo tee` 😄 because the installer is a live ISO and you run it with root user, command by command.

😄😄😄😄😄

peterbabic commented:
@bl-ue no poblem as well! Thanks for suggestions. Did not know about the nature of "an ef-es-tab file" 😄

bl-ue commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+
+`genfstab -U {{path/to/mount_point}}`
+
+- Generate an fstab file, requires being a superuser:
```
Perhaps this will read better?
```suggestion
- Generate an fstab file, requires root permissions:
```

bl-ue commented on pages/linux/genfstab.md:
```diff
@@ -0,0 +1,20 @@
+> Generate output suitable for addition to an fstab file.
+> More information: <https://man.archlinux.org/man/extra/arch-install-scripts/genfstab.8>.
+
+- Generate the output based on a label:
```
Perhaps this will read better?
```suggestion
- Generate output based on a label:
```

bl-ue reviewed (changes requested):
Okay, thanks for the help @navarroaxel.  @peterbabic I have only two more comments.

peterbabic commented:
I have added another commit where I believe the changes are the most consistent. I have also removed backticks around `fstab` as it is word, not a code nor a path.

peterbabic commented:
Maybe the last thing I would check is if *volume* is the right word, as *filesystem*, *block device*, *disk*, *partition* or *disk partition* would all suit as well depending on use case.

 I would keep *volume* there.

![image](https://user-images.githubusercontent.com/3952052/106497597-8bfb5400-64be-11eb-9062-a7e45dfe73c2.png)

bl-ue commented:
@peterbabic - Looks good to me, except the sentence "A usual way to generate an fstab file, requires root permissions:" is inconsistent with the rest of our pages.  If that is the most common example, then I think you should:
1. Move the example to the top, so it's what the user sees first, and
2. Change the example description to "Generate an fstab file (requires root permissions):"

sbrl reviewed (approved):
Looks pretty much good to go to me - I think we can probably merge this.

I've left a comment on the `requires root permissions` issue, but that doesn't necessarily need to block the merge of this PR - and I can see some significant discussion has already happened - so I'm leaning on the side of getting this merged sooner rather than later.

bl-ue commented:
@sbrl that was actually already resolved :) - https://github.com/tldr-pages/tldr/pull/5215#discussion_r569025992

bl-ue reviewed (approved):
Thank you @peterbabic, and welcome to tldr-pages!

sbrl commented:
Ok, whatever was done there is fine then :-)

bl-ue commented:
We picked not to use `sudo` because that command is normally run in a boot environment where `tee` may not be available.
